### PR TITLE
Issue 16 17 18

### PR DESCRIPTION
**Pull request by wwllaad**

### CODE REVIEW

File: src/main/java/com/sevgmo/stationpassport/StationPassportApplication.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 
-
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+
 @SpringBootApplication
 public class StationPassportApplication {
 
+
```
Comment (lantushenkoao) [inline src/main/java/com/sevgmo/stationpassport/StationPassportApplication.java]: Remove redundant blank lines

File: src/main/java/com/sevgmo/stationpassport/model/Section.java [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
     }
 
     public void setName(String name) {
+
         this.name = name;
     }
 
-    public Integer getParentSectionId() {
-        return parentSectionId;
+    public Section getSection() {
```
Comment (lantushenkoao) [inline src/main/java/com/sevgmo/stationpassport/model/Section.java]: rename to getParent()

File: src/main/java/com/sevgmo/stationpassport/serialize/CustomFieldDTO.java [hunk 3]
```diff
@@ -3,29 +3,52 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import com.sevgmo.stationpassport.model.AbstractEntity;
 import com.sevgmo.stationpassport.model.CustomField;
+import com.sevgmo.stationpassport.model.Section;
 
 @JsonSerialize
 public class CustomFieldDTO extends AbstractEntity {
 
-    private int sectionId;
+    private Section section;
```
Comment (lantushenkoao) [inline src/main/java/com/sevgmo/stationpassport/serialize/CustomFieldDTO.java]: DTO classes should not have domain objects as fields

File: src/main/java/com/sevgmo/stationpassport/service/StationService.java [hunk 4]
```diff
@@ -47,15 +47,19 @@ public CustomFieldDTO getCustomFieldDTOById(int id){
         return customFieldDTO;
     }
 
+    @JsonSerialize
+    public CustomField getCustomFieldById(int id){
```
Comment (lantushenkoao) [inline src/main/java/com/sevgmo/stationpassport/service/StationService.java]: Do not serialize domain objects directly. Use DTOs

File: src/main/java/com/sevgmo/stationpassport/service/StationService.java [hunk 5]
```diff
@@ -84,7 +88,7 @@ public StationApiFormDTO getStationApiDTOById(int id) {
 
         //Loading CustomFieldDTO
         for(CustomFieldValueDTO customFieldValueDTO : customFieldValueDTOList){
```
Comment (lantushenkoao) [inline src/main/java/com/sevgmo/stationpassport/service/StationService.java]: Consider using array utility methods, like
customFieldValueDTOList.forEach

File: src/main/java/com/sevgmo/stationpassport/service/StationService.java [hunk 6]
```diff
@@ -94,7 +98,7 @@ public StationApiFormDTO getStationApiDTOById(int id) {
         int sectionIdCounter = 0;
         for(CustomFieldDTO customFieldDTO : customFieldDTOList){
```
Comment (lantushenkoao) [inline src/main/java/com/sevgmo/stationpassport/service/StationService.java]: Consider using Java8 streams here, like customFieldDTOList.stream()....

**Review by lantushenkoao — CHANGES_REQUESTED** [review]

**Comment by wwllaad** [comment]
Requested changes were made.

**Pull request merged**